### DISCUSSION TITLE
Dependencies/Dependents

### DISCUSSION THREAD

**Discussion by mfasman-ecolab** [vercel/turborepo#9655] [Help]
### Summary

I am trying to figure out why the `--affected` flag doesn't seem to be working for me, and in the process of going down this rabbit hole I have come across the dependencies and dependents output that is a part of `--dry-run` runs of turbo.

The screenshot below is for a library that should both depend on several others in my repo - both dependencies and devDependencies - and be depended on by an application in my library. The command being run is `turbo test --affected --dry-run`.

I have two questions:
- What are the dependencies and dependents in the screenshot below supposed to reflect? Do they have anything to do with how `--affected` is supposed to work?
- Why is `--affected` always resulting in all of my packages running their task, instead of only running the packages that have changed or have a dependency that has changed?

![Screenshot 2024-12-27 at 10 02 58 PM](https://github.com/user-attachments/assets/f594271f-af04-4a36-8ecd-0c37f2a24a18)

### Additional information

- I am using pnpm and my pnpm workspace file has been sufficient for all other tools that rely on it
- I have also tried duplicating my workspace configuration into my root `package.json`. I see no difference in behavior with or without this change.
- I have set `turbo` to `latest` in my `package.json` and am using `pnpm@9.15.2`, which is latest at time of writing
- My dependencies inside the monorepo are expressed using the `pnpm` convention of `"workspace:*"`, e.g. `"@oneapp/oneapp-example-feature": "workspace:*"`

### Example

I am working in a private repo so I cannot link to it. I could try to create a minimum reproducible example but I wouldn't even know what parts of the project would be relevant to causing this issue.

**Comment by anthonyshew** [comment]
"Dependencies" refers to the set of packages that the package depends on. "Dependents" are the reverse, the set of packages that depend on that package. As an example, in `create-turbo`, `@repo/ui` is a _dependency_ of `web` and `web` is a _dependent_ of `@repo/ui`.

As for `--affected` always running all tasks, it could be because history is too shallow, as documented in [the warning found here](https://turbo.build/repo/docs/reference/run#--affected). It's hard to say if this is the case for your repository without a reproduction, but sounds to be a likely culprit.

**Reply by mfasman-ecolab** [reply]
Thanks for the response! That's what I assumed those terms would mean, but the thing that confuses me is that the pictured package has a dependency in the monorepo - `"@oneapp/routing": "workspace:*"` - and is itself a dependency of another package in the monorepo. Yet it doesn't list those two examples or any of the others I'd expect to have seen there. This is part of why I wondered if it might relate to why `--affected` was not working as expected, since it does not seem to be recognizing its dependency or dependent relations.

I have also done nothing that would result in the repository being a shallow clone. Just to confirm I also ran the following command `git rev-parse --is-shallow-repository` and the result was `false`.
